src/typ.rs

```diff
@@ -1341,6 +1425,33 @@ pub fn infer_module(
                 })
             }
 
+            Statement::TypeAlias {
+                meta,
+                public,
+                alias,
+                real_type,
+            } => {
+                let mut type_vars = hashmap![];
```

Comment:
I think we should check that the type being aliased exists before we register it. i.e. this should be an error:

```rust
type MyAlias = TypeThatDoesNotExist
```

Reply:
this is already handled, I will only `insert_type_alias` if and only if the resolved type is legal, otherwise it will be an unknown type error

![image](https://user-images.githubusercontent.com/6234553/74090042-1abed600-4ae2-11ea-91ce-a6cbb8a7cbf7.png)

Reply:
Ah yes! I forgot that function does that. Thank you :)